crypto/async/async.c

```diff
@@ -203,7 +203,9 @@ int ASYNC_start_job(ASYNC_JOB **job, ASYNC_WAIT_CTX *wctx, int *ret,
             }
 
             if (ctx->currjob->status == ASYNC_JOB_PAUSED) {
-                ctx->currjob = *job;
+                if ((ctx->currjob = *job) == NULL)
```

Comment:
I'm not even sure that this line is needed. It looks redundant to me - but certainly if this is NULL then something has gone badly wrong. We should error out.

Reply:
Line 184 has a check for *job == NULL which is why Coverity was complaining about a subsequent dereference.  I.e. it looks like this case was planned for but not entirely handled.

Reply:
This is a "should not happen" scenario. If we get here with `*job` being NULL, then it means `ctx->currjob` was already non-NULL and the status is `ASYNC_JOB_PAUSED`. But the only place where we set the status to `ASYNC_JOB_PAUSED` is a few lines above and we immediately set `ctx->currjob` to NULL immediately afterwards and return from the function. So, AFAICT, the only we can get here is if `*job` is non-NULL, and the assignment on line 185 has already taken place. But if that is the case then the assignment being done here appears redundant.

I'm a a little wary of removing it completely (although the async test still passes if you do so), but I would make this an error and/or assert.

Reply:
Okay, I've reverted the code and added a *job == NULL check that returns an error.